lib/plugins/utils.nix

```diff
@@ -110,5 +123,24 @@
           path = loc;
         };
       };
+
+      docs.options.${docsfile} = {
+        title = lib.last loc;
+        description = lib.concatLines (
+          [
+            "**URL:** [${url}](${url})"
+            ""
+          ]
+          ++ lib.optionals (maintainers != [ ]) [
+            "**Maintainers:** ${maintainersString}"
+            ""
+          ]
+          ++ lib.optionals (description != null && description != "") [
+            "---"
+            ""
+            description
+          ]
+        );
+      };
```

Comment:
As a gut reaction it feels a bit strange to have a function called `mkMetaModule` that sets other things than meta arguments.
It feels a "duplicate" of the metadata in another form, can't we extract it with a well placed library function? (Not entirely sure, as the metadata is a bit of a complicated beast --')

Reply:
There is a lot of deliberately duplicated stuff, with the idea being that the "old" way will eventually be removed.